subtrie/src/node.rs

```diff
@@ -263,17 +270,18 @@ pub enum NodeOwned<H, L> {
 	/// Null trie node; could be an empty root or an empty branch entry.
 	Empty,
 	/// Leaf node; has key slice and value. Value may not be empty.
-	Leaf(NibbleVec, ValueOwned<H, L>),
+	Leaf(NibbleVec, ValueOwned<H, L>, L),
```

Comment:
What's the reason for storing the location of the node here? I could not find where this is actually used.

Reply:
Took me a while to remember, but I think this location is an additional one that is optional on branch and leaf, it holds the location of a child trie root in db (so it does not get overwritten on update (child trie location injection is in substrate iirc, but if the node is touched  then the db will overwrite this n + 1 location so it had to be loaded in node to be set again unchanged).